torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -391,6 +392,16 @@ std::string GetLiveTensorsReport(size_t nodes_threshold,
   return ss.str();
 }
 
+static std::string GetLiveTensorsStableHLO(
+    const std::string& device_str, const std::vector<std::string>& devices) {
+  tsl::profiler::TraceMe activity("GetLiveTensorsStableHLO",
+                                  tsl::profiler::TraceMeLevel::kInfo);
+  torch::lazy::BackendDevice device = GetDeviceOrCurrent(device_str);
+  auto tensors = XLAGraphExecutor::Get()->GetLiveTensors(&device);
+  return XLAGraphExecutor::Get()->SyncTensorsGraphDumpHlo(
```

Comment:
I want to make sure my understanding on "Live Tensors" and "Sync" is correct. Please correct if I'm wrong

"Live Tensors" refers to all tensors that are already on device, including both materialized and unmaterialized tensors.

"Sync" refers to materialize the IR Nodes that are not on the device yet. It involves XLA compilation and graph execution.

This API is built for the use case that user would like to get the StableHLO of the whole graph. It's like a `mark_step()` but doesn't compile and execute the graph.

Since `SyncTensorsGraphDumpHlo` introduces a new workflow that returns early before XLA compilation and graph execution. This expands the definition of "Sync" above from "materialize IR Nodes" to "Lower the IR Nodes"

Reply:
"Live Tensors" means all xla tensor that are currently live in the Python space. It doesn't matter whether they are device data or IR or aten tensors. I think that's all you need and you don't need to sync them. Can you try just passing the live tensors to the hlodump and see if that gives you what you wants?